File: Example/XenditExampleObjC/XenditExampleObjC/OpenSansFont.m
#import "OpenSansFont.h"
#import <CoreText/CoreText.h>

@implementation OpenSansFont

+ (void)register {
    NSArray<NSString *> *names = @[
        @"OpenSans-Regular",
        @"OpenSans-Medium",
        @"OpenSans-SemiBold",
        @"OpenSans-Bold",
    ];
    for (NSString *name in names) {
        NSURL *url = [[NSBundle mainBundle] URLForResource:name withExtension:@"ttf"];
        if (url) {
            CTFontManagerRegisterFontsForURL((__bridge CFURLRef)url, kCTFontManagerScopeProcess, NULL);
        } else {
            NSLog(@"[XenditExampleObjC] '%@.ttf' not found in bundle — add it to Copy Bundle Resources in Xcode.", name);
        }
    }
}

+ (UIFont *)regular  { return [UIFont fontWithName:@"OpenSans-Regular"  size:14] ?: [UIFont systemFontOfSize:14 weight:UIFontWeightRegular]; }
+ (UIFont *)medium   { return [UIFont fontWithName:@"OpenSans-Medium"   size:14] ?: [UIFont systemFontOfSize:14 weight:UIFontWeightMedium]; }
+ (UIFont *)semiBold { return [UIFont fontWithName:@"OpenSans-SemiBold" size:14] ?: [UIFont systemFontOfSize:14 weight:UIFontWeightSemibold]; }
+ (UIFont *)bold     { return [UIFont fontWithName:@"OpenSans-Bold"     size:14] ?: [UIFont systemFontOfSize:14 weight:UIFontWeightBold]; }

@end

File: Example/XenditExampleUIKit/XenditExampleUIKit/OpenSansFont.swift
import CoreText
import UIKit

enum OpenSansFont {

    static func register() {
        let names = [
            "OpenSans-Regular",
            "OpenSans-Medium",
            "OpenSans-SemiBold",
            "OpenSans-Bold",
        ]
        for name in names {
            guard let url = Bundle.main.url(forResource: name, withExtension: "ttf") else {
                print("[XenditExampleUIKit] '\(name).ttf' not found in bundle — add it to Copy Bundle Resources in Xcode.")
                continue
            }
            CTFontManagerRegisterFontsForURL(url as CFURL, .process, nil)
        }
    }

    static var regular:  UIFont { UIFont(name: "OpenSans-Regular",  size: 14) ?? .systemFont(ofSize: 14, weight: .regular) }
    static var medium:   UIFont { UIFont(name: "OpenSans-Medium",   size: 14) ?? .systemFont(ofSize: 14, weight: .medium) }
    static var semiBold: UIFont { UIFont(name: "OpenSans-SemiBold", size: 14) ?? .systemFont(ofSize: 14, weight: .semibold) }
    static var bold:     UIFont { UIFont(name: "OpenSans-Bold",     size: 14) ?? .systemFont(ofSize: 14, weight: .bold) }
}
